edb/schema/lproperties.py

```diff
@@ -311,21 +375,37 @@ class AlterProperty(
     referenced_astnode = qlast.AlterConcreteProperty
 
     @classmethod
-    def _cmd_tree_from_ast(cls, schema, astnode, context):
+    def _cmd_tree_from_ast(
+        cls,
+        schema: s_schema.Schema,
+        astnode: qlast.DDLOperation,
+        context: sd.CommandContext,
+    ) -> referencing.AlterReferencedInheritingObject[Property]:
         cmd = super()._cmd_tree_from_ast(schema, astnode, context)
 
         if isinstance(astnode, qlast.CreateConcreteProperty):
+            assert isinstance(cmd, pointers.PointerCommand)
             cmd._process_create_or_alter_ast(schema, astnode, context)
 
+        assert isinstance(cmd, referencing.AlterReferencedInheritingObject)
         return cmd
 
-    def _apply_field_ast(self, schema, context, node, op):
+    def _apply_field_ast(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+        node: qlast.DDLOperation,
+        op: sd.AlterObjectProperty,
+    ) -> None:
+        # TODO: this method is not covered by tests
```

Comment:
@elprans, this function is not covered by tests. Can you please give me an hint, on what kind of test should be arranged to cover this scenario? What is `op.new_value` in this context?

Reply:
Interesting.  This a deparsing of an `ALTER PROPERTY SET TYPE <foo>`.  To cover this, add a test to `TestGetMigration` that changes the type of a property. `new_value` in this case is `types.Type`.

Reply:
I did several attempts to configure a test, following the same patterns already defined in `TestGetMigration` methods, but it looks like I am not arranging the right test. I confirm that the method is not hit by current suite.

Can you show me how to apply a second migration to a schema already modified by a first migration in code?

Reply:
Yes, we seem to have a fairly large coverage gap. I'm working on fixing the hardness in `TestGetMigration` to support testing multi-step migration deparse.

Reply:
The enabled tests are a much deeper rabbit hole than I anticipated.  Let's not block this PR on that.

Reply:
Thanks! I was going to propose the same.